ps_crossselling.php

```diff
@@ -246,7 +246,8 @@ protected function getOrderProducts(array $productIds = [])
         FROM ' . _DB_PREFIX_ . 'orders o
         LEFT JOIN ' . _DB_PREFIX_ . 'order_detail od ON (od.id_order = o.id_order)
         WHERE o.valid = 1
-        AND od.product_id IN (' . implode(',', $productIds) . ')';
+        AND od.product_id IN (' . implode(',', $productIds) . ')
+        ORDER BY date_add desc LIMIT ' . ((int) Configuration::get('CROSSSELLING_NBR')) * 50;
```

Comment:
IMHO pagination size should use a variable with a default value defined as class constant:
```
protected function getOrderProducts(array $productIds = [], int $size = self::ORDER_PRODUCTS_SIZE)
```
```suggestion
        ORDER BY date_add desc LIMIT ' . ((int) Configuration::get('CROSSSELLING_NBR')) * $size;
```